src/mixins/FetchFacesMixin.js

```diff
@@ -120,14 +117,19 @@ export default {
 					}
 				)
 
-				fetchedFiles = fetchedFiles
-					.map(file => genFileInfo(file))
-					.map(file => ({ ...file, filename: he.decode(file.realpath).replace(`/${getCurrentUser().uid}/files`, `/files/${getCurrentUser().uid}`) }))
-					.map(file => ({ ...file, faceDetections: JSON.parse(he.decode(file.faceDetections)) }))
-
-				const fileIds = fetchedFiles.map(file => '' + file.fileid)
-
-				this.appendFiles(fetchedFiles)
+				const processedFiles = [];
+				const fileIds = [];
+			
+				for (const file of fetchedFiles) {
+				  const processedFile = genFileInfo(file);
+				  processedFile.filename = he.decode(processedFile.realpath).replace(`/${getCurrentUser().uid}/files`, `/files/${getCurrentUser().uid}`);
+				  processedFile.faceDetections = JSON.parse(he.decode(processedFile.faceDetections));
+			
+				  processedFiles.push(processedFile);
+				  fileIds.push(String(processedFile.fileid));
+				}
+			
+				this.appendFiles(processedFiles);
```

Comment:
I think it's a nice change. Not sure if strictly necessary but surely improves readability a bit I think.

Reply:
Thank you!